services/notify/src/api.js

```diff
@@ -202,7 +201,7 @@ builder.declare({
 
   await req.authorize(req.body);
   try {
-    await this.DenylistedNotification.remove(address);
+    await this.db.fns.delete_denylist_address(address);
   } catch (e) {
     if (e.name !== 'ResourceNotFoundError') {
```

Comment:
If row does not exist, we could have `delete_denylist_address` throw error code `P0002` with error message 'no such row' then this line could be changed to:

```
if (e.code !== 'P0002') {
```

https://github.com/taskcluster/taskcluster/search?q=P0002&unscoped_q=P0002

Reply:
Ah yes, good point!

Reply:
It looks like the JS catches that error just to ignore it, much like adding a row, in order to achieve idempotency.  So I think the same logic applies: if the delete just ignores the case when the row doesn't exist, then that's simpler SQL and simpler JS.